File: Functions/computeStrippingMetric.m
function metric = computeStrippingMetric(seg, windowS)
% computeStrippingMetric - quantitative Li-stripping indicator for one
% post-charge C/5 discharge segment (A-002).
%
% Definition: max(dV/dt) - min(dV/dt) over the early-discharge time window
% windowS = [t1 t2] (seconds, relative to segment start). A smooth (no-
% stripping) curve has near-constant dV/dt over this window -> small
% metric. A Li-stripping segment shows a plateau (voltage roughly flat for
% a while) which produces a localised bump in dV/dt -> large metric.
%
% Inputs:
%   seg     - struct produced by extractDVdtSegmentsAll (uses .timeS_interp
%             and .dVdt_Vpers)
%   windowS - 2-element vector [t_start t_end] in seconds
%
% Output:
%   metric  - scalar [V/s], NaN if the window has too few samples
%
% Author: GitHub Copilot (for Feye Hoekstra)
% Date:   2026-07-28 (extracted from PlotCellSummary.m into a shared helper)

metric = NaN;
if isempty(seg) || ~isfield(seg, 'dVdt_Vpers') || ~isfield(seg, 'timeS_interp')
    return
end
t = seg.timeS_interp;
y = seg.dVdt_Vpers;
m = (t >= windowS(1)) & (t <= windowS(2)) & ~isnan(y);
if nnz(m) < 5
    return
end
metric = max(y(m)) - min(y(m));
end

File: Functions/computeStrippingMetric.py
"""
computeStrippingMetric - quantitative Li-stripping indicator for one
post-charge C/5 discharge segment (A-002): max(dV/dt) - min(dV/dt) over an
early-discharge time window.

Python counterpart of Functions/computeStrippingMetric.m.

Author: GitHub Copilot (for Robinson Medina / Feye Hoekstra)
Date: 2026-08-24
"""

import numpy as np


def compute_stripping_metric(seg, window_s):
    """Return max(dV/dt) - min(dV/dt) over window_s=[t1,t2] for one segment dict."""

    if not seg or 'dVdt_Vpers' not in seg or 'timeS_interp' not in seg:
        return np.nan
    t = seg['timeS_interp']
    y = seg['dVdt_Vpers']
    m = (t >= window_s[0]) & (t <= window_s[1]) & ~np.isnan(y)
    if np.count_nonzero(m) < 5:
        return np.nan
    return np.max(y[m]) - np.min(y[m])


__all__ = ['compute_stripping_metric']
